### PR TITLE
Update MacroRecorderPostprocessor tests

### PR DESCRIPTION
**Pull request by hinerm**
This fixes https://github.com/imagej/imagej-legacy/issues/305 for me

### CODE REVIEW

**Comment by hinerm** [comment]
@ctrueden can you check if this works on your systems?

File: src/main/java/net/imagej/legacy/DefaultLegacyHooks.java [hunk 1]
```diff
@@ -357,7 +351,7 @@ public void addMenuItem(final String menuPath, final String command) {
 						final String mPath = m.group(1);
 						final String label = m.group(2);
 						final String plugin = m.group(3);
-						helper.addCommand(mPath, label, plugin);
+						legacyService.getIJ1Helper().addCommand(mPath, label, plugin);
```
Comment (ctrueden) [inline src/main/java/net/imagej/legacy/DefaultLegacyHooks.java:354]: Can't you just continue using the `helper` variable here? If not, then isn't there a chance that `legacyService.getIJ1Helper()` will now return null, yielding an NPE here?

**Review by ctrueden — COMMENTED** [review]
All tests pass with this branch on my Linux system!

I see that all references to the `helper` variable have changed to access it dynamically from the `LegacyService` every time. But there is still a null check in at least one spot. Is the idea that the `IJ1Helper` can go from null to non-null or vice versa? If so, what if it happens in between the null check and the next dynamic access?

File: src/main/java/net/imagej/legacy/DefaultLegacyHooks.java (hunk 1 above)
Comment (hinerm) [reply src/main/java/net/imagej/legacy/DefaultLegacyHooks.java:354]: @ctrueden there are some `helper` local variable instances when the helper is used multiple times in a given scope, but not here. Could cache it outside of the loop though if you think that's worthwhile?

It's true that the `IJ1Helper` could be set to `null` at some point, but I just noticed that the [`setter` is documented as non-API](https://github.com/imagej/imagej-legacy/blame/915bec8e5776a348bd5481d76d40b3c98bd64227/src/main/java/net/imagej/legacy/LegacyService.java#L610-L616). So it should never actually be called in production. Therefore the [existing `null` check](https://github.com/imagej/imagej-legacy/blob/915bec8e5776a348bd5481d76d40b3c98bd64227/src/main/java/net/imagej/legacy/DefaultLegacyHooks.java#L151) seems superfluous and could be removed - would you agree?

Comment (ctrueden) [reply src/main/java/net/imagej/legacy/DefaultLegacyHooks.java:354]: Yeah, the null check looks unnecessary to me. Looking at the `LegacyService` code, it's convoluted that the `IJ1Helper` instance gets passed as a second argument to the `DefaultLegacyHooks`, when it's already being given the `LegacyService` that can provide access to that very same object. So I like your changes. What I'd suggest is to add a new method to `DefaultLegacyHooks` like this:

```java
private IJ1Helper helper() { return legacyService.getIJ1Helper(); }
```

And optionally do some null check in there, if we want to be super defensive.

But regardless, then we can just change all the `helper.doSomething` codes to `helper().doSomething`, which will still read nicely.

What do you think?

**Pull request merged**